cpp/example_code/dynamodb/scan_table.cpp

```diff
@@ -1,102 +1,122 @@
-        "Where:\n"
-        "    table - the table to Scan.\n\n"
-        "You can add an optional projection expression (a quote-delimited,\n"
-        "comma-separated list of attributes to retrieve) to limit the\n"
-        "fields returned from the table.\n\n"
-        "Example:\n"
-        "    scan_table HelloTable \n"
-        "    scan_table SiteColors \"default, bold\"\n";
-
-    if (argc < 2)
-    {
-        std::cout << USAGE;
+#include "dynamodb_samples.h"
+
+// snippet-start:[dynamodb.cpp.scan_table.code]
+//! Scan a DynamoDB table.
+/*!
+  \sa scanTable()
+  \param tableName: Name for the DynamoDB table.
+  \param projectionExpression: An optional projection expression, ignored if empty.
+  \param clientConfiguration: AWS client configuration.
+  \return bool: Function succeeded.
+ */
+
+bool AwsDoc::DynamoDB::scanTable(const Aws::String &tableName,
+                                 const Aws::String &projectionExpression,
+                                 const Aws::Client::ClientConfiguration &clientConfiguration) {
+    Aws::DynamoDB::DynamoDBClient dynamoClient(clientConfiguration);
+    Aws::DynamoDB::Model::ScanRequest request;
+    request.SetTableName(tableName);
+
+    if (!projectionExpression.empty())
+        request.SetProjectionExpression(projectionExpression);
+
+    // Perform scan on table.
+    const Aws::DynamoDB::Model::ScanOutcome &outcome = dynamoClient.Scan(request);
+    if (outcome.IsSuccess()) {
+        // Reference the retrieved items.
+        const Aws::Vector<Aws::Map<Aws::String, Aws::DynamoDB::Model::AttributeValue>> &items = outcome.GetResult().GetItems();
+        if (!items.empty()) {
+            std::cout << "Number of items retrieved from scan: " << items.size()
+                      << std::endl;
+            //Iterate each item and print.
+            for (const Aws::Map<Aws::String, Aws::DynamoDB::Model::AttributeValue> &itemMap: items) {
+                std::cout << "******************************************************"
+                          << std::endl;
+                // Output each retrieved field and its value.
+                for (const auto &itemEntry: itemMap)
+                    std::cout << itemEntry.first << ": " << itemEntry.second.GetS()
+                              << std::endl;
+            }
+        }
+
+        else {
+            std::cout << "No item found in table: " << tableName << std::endl;
+        }
+    }
+    else {
+        std::cerr << "Failed to Scan items: " << outcome.GetError().GetMessage()
+                  << std::endl;
+    }
+
+    return outcome.IsSuccess();
+}
+
+// snippet-end:[dynamodb.cpp.scan_table.code]
+
+/*
+ *  main function
+ *
+ *  Usage: 'run_scan_table <table> [projection_expression]'
+ *
+ *  Prerequisites: A pre-populated DynamoDB table.
+ *
+ *  Instructions for populating a table with sample data can be found at:
+ *  https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SampleData.html

```

Comment:
*  For instructions on populating a table with sample data, see
 *  https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SampleData.html.